### PR TITLE
Request behaviour

### LINKED ISSUE #96: Implement request behaviour
ℹ️ This issue is part of Epic #62 

## Context

This behaviour will establish the default functions that will be used by the request module

## Objectives

* Implements a default behaviour for the request modules:
    `prepare`: input a CommandPayload and a keyset, and it will return a request_body struct
    `process`: input a request_body and returns a response_body struct

### PR DESCRIPTION
**Pull request by boterop**
## Description

Fixes #96 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: lib/chainweb/pact/spec.ex [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    SendResponse,
+    SPVRequestBody,
+    SPVResponse
+  }
+
+  @type listen_request :: ListenRequestBody.t()
+  @type local_request :: LocalRequestBody.t()
+  @type poll_request :: PollRequestBody.t()
+  @type send_request :: SendRequestBody.t()
+  @type spv_request :: SPVRequestBody.t()
```
Comment (einerzg) [inline lib/chainweb/pact/spec.ex]: IMO, you don't need to define types for this. Better use this like this:

`@type request :: ListenRequestBody.t() | LocalRequestBody.t() | PollRequestBody.t() | SendRequestBody.t() | SPVRequestBody.t()`

File: lib/chainweb/pact/spec.ex [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+  @type listen_response :: ListenResponse.t()
+  @type local_response :: LocalResponse.t()
+  @type poll_response :: PollResponse.t()
+  @type send_response :: SendResponse.t()
+  @type spv_response :: SPVResponse.t()
```
Comment (einerzg) [inline lib/chainweb/pact/spec.ex]: the same for this code!!!!

**Review by einerzg — CHANGES_REQUESTED** [review]

File: lib/chainweb/pact/spec.ex [hunk 3]
```diff
@@ -0,0 +1,47 @@
+defmodule Kadena.Chainweb.Pact.Spec do
```
Comment (juanhurtado10) [inline lib/chainweb/pact/spec.ex]: @santiagoboterokommit please, rename this module according to its context. IMO, `Spec` does not make sense and it is not a default name for contacts specification

File: lib/chainweb/pact/request.ex [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    LocalResponse,
+    PollRequestBody,
+    PollResponse,
+    SendRequestBody,
+    SendResponse,
+    SPVRequestBody,
+    SPVResponse
+  }
+
+  @type network :: :public | :test | nil
```
Comment (einerzg) [inline lib/chainweb/pact/request.ex]: It's `nil` a network type valid for Kadena? please remove

**Review by einerzg — CHANGES_REQUESTED** [review]

File: lib/chainweb/pact/request.ex (hunk 4 above)
Comment (boterop) [reply lib/chainweb/pact/request.ex]: srry my bad, fixed now

**Review by einerzg — APPROVED** [review]

**Pull request merged**